types/tx_msg.go

```diff
@@ -45,15 +47,20 @@ type Tx interface {
 	// invalid), then the corresponding signature is
 	// .Empty().
 	GetSignatures() []StdSignature
+
+	GetSequence() int64
 }
 
 var _ Tx = (*StdTx)(nil)
 
 // StdTx is a standard way to wrap a Msg with Signatures.
 // NOTE: the first signature is the FeePayer (Signatures must not be nil).
+// NOTE: a single Sequence is included in the StdTx, separate from the Msg
+// 	- it must be incldued in the SignBytes to actually provide replay protection
 type StdTx struct {
 	Msg
 	Signatures []StdSignature
+	Sequence   int64
```

Comment:
We don't need this here, it's implied by the first signature.

Reply:
I think this should be in the StdTx instead of StdSignature so we don't have to add an optional Sequence field to StdSignature that only gets added in the first StdSignature but not the rest.